ouroboros-consensus/src/consensus-testlib/Test/Ouroboros/Consensus/ChainGenerator/RaceIterator.hs

```diff
@@ -13,6 +13,22 @@ module Test.Ouroboros.Consensus.ChainGenerator.RaceIterator (
   , nextConservative
   ) where
 
+{- | These functions iteratively produce all race windows in a slot vector.
+
+The first window is produced by 'init', which unconditionally starts the window at the first slot.
+This window can then be passed to 'next', which starts the new window after the first active slot.
+
+@
+---X--X--X--X
+^ start of window 1 from 'init'
+    ^ start of window 2 from 'next'
+       ^ start of window 3 from 'next'
+@
+
+Valid windows must have @k+1@ active slots.
+If the vector doesn't have sufficient slots to meet this condition, 'init' and 'next' return 'Nothing' and we fall back
+to 'initConservative' and 'nextConservative', which pad the window with only active slots.
```

Comment:
What do you mean by "pad the window with only active slots"? Also see https://github.com/input-output-hk/ouroboros-consensus/pull/240#discussion_r1295818408

Reply:
oops, wrote that from the wrong perspective

Reply:
> If the vector doesn't have sufficient slots to meet this condition, `init` and `next` return `Nothing` and we fall back to `initConservative` and `nextConservative`, which create a window of size `scg - delta` unconditionally, which is later filled in with active slots to have maximum density within the chain growth limit.

Hmm, I am still confused, where do we later fill conservative race windows with active slots of maximum density? For comparison, WDYT about this:

> If the vector doesn't have sufficient slots to meet this condition, `init` and `next` return `Nothing` and we fall back to `initConservative` and `nextConservative`, which create a window of size `scg - delta` unconditionally, in order to constrain adversarial active slots there to satisfy the Length of Competing Chains Assumption.

Reply:
hmmm right I interpreted this:
```
the vector length spans the full stability window (minus delta) and has all the exceeding slots active.
```

and your
```
 as sparse as still allowed by chain growth
```

as implying that `uniformAdversarialChain` treats the spillover part specially, but it doesn't